Consul/Catalog.cs

```diff
@@ -211,12 +212,26 @@ public Task<QueryResult<CatalogService[]>> Service(string service, string tag, C
         /// <param name="ct">Cancellation token for long poll request. If set, OperationCanceledException will be thrown if the request is cancelled before completing</param>
         /// <returns>A list of service instances</returns>
         public Task<QueryResult<CatalogService[]>> Service(string service, string tag, QueryOptions q, CancellationToken ct)
+        {
+            return Service(service, new[] {tag}, q, ct);
+        }
+        
+        /// <summary>
+        /// Service is used to query catalog entries for a given service
+        /// </summary>
+        /// <param name="service">The service ID</param>
+        /// <param name="tags">A tag to filter on</param>
+        /// <param name="q">Customized query options</param>
+        /// <param name="ct">Cancellation token for long poll request. If set, OperationCanceledException will be thrown if the request is cancelled before completing</param>
+        /// <returns>A list of service instances</returns>
+        public Task<QueryResult<CatalogService[]>> Service(string service, IList<string> tags, QueryOptions q, CancellationToken ct)
         {
             var req = _client.Get<CatalogService[]>(string.Format("/v1/catalog/service/{0}", service), q);
-            if (!string.IsNullOrEmpty(tag))
+            if (tags?.Any(s => !string.IsNullOrEmpty(s)) == true)
```

Comment:
I think it would be nice to drop empty tags here, not adding everything to the request.